include/netlink/attr.h

```diff
@@ -118,6 +118,10 @@ extern int64_t          nla_get_s64(const struct nlattr *);
 extern int              nla_put_s64(struct nl_msg *, int, int64_t);
 extern uint64_t		nla_get_u64(const struct nlattr *);
 extern int		nla_put_u64(struct nl_msg *, int, uint64_t);
+extern int64_t		nla_get_sint(const struct nlattr *);
+extern int              nla_put_sint(struct nl_msg *, int, int64_t);
+extern uint64_t		nla_get_uint(const struct nlattr *);
+extern int              nla_put_uint(struct nl_msg *, int, uint64_t);
```

Comment:
doesn't this lack a `NLA_UINT` enum value (beside `NLA_U32`)?

Note that so far, the enum values (like `NLA_U32`) in `include/netlink/attr.h` mirror the ones from kernel's `include/net/netlink.h`. I think, you should (possibly in a separate commit) copy the enum over from the kernel header, so that they are the same numeric values. Even if it (probably?) would not matter if they have different values...

Maybe NLA types that are not implemented by libnl (e.g. `NLA_BITFIELD32`, `NLA_REJECT`) should be excluded, however, ensure that the numeric values of `NLA_UINT` is the same as the one in kernel.

Then also adjust things like `nla_attr_minlen`.

Reply:
Thanks for pointing out. Will do: 1) define NLA_SINT/UINT as values in kernel's include/net/netlink.h. 2) adjust nla_attr_minlen().